app/service/EmailPreferencesService.scala

```diff
@@ -19,101 +19,100 @@ package service
       apis              <- getOrUpdateFlowWithVisibleApis(existingFlow, developerSession)
-      visibleCategories = apis.map(_.categories).reduce(_ ++ _).distinct.sorted
-      categories        <- fetchAllAPICategoryDetails().map(_.filter(x => visibleCategories.contains(x.category)))
-    } yield categories
+      visibleCategoryNames = apis.map(_.categories).reduce(_ ++ _).distinct.map(_.value).sorted
+      categories        <- fetchAllAPICategoryDetails().map(_.filter(x => visibleCategoryNames.contains(x.category)))
+    } yield categories.distinct.sortBy(_.category)
 
 
-  private def getOrUpdateFlowWithVisibleApis(existingFlow: EmailPreferencesFlow, developerSession: DeveloperSession)
-                                            (implicit hc: HeaderCarrier): Future[List[ApiDefinition]] = {
+  private def getOrUpdateFlowWithVisibleApis(existingFlow: EmailPreferencesFlowV2, developerSession: DeveloperSession)
+                                            (implicit hc: HeaderCarrier): Future[List[CombinedApi]] = {
     NonEmptyList.fromList(existingFlow.visibleApis.toList).fold({
-      val visibleApis = apmConnector.fetchApiDefinitionsVisibleToUser(developerSession.developer.userId)
-      updateVisibleApis(developerSession, apmConnector.fetchApiDefinitionsVisibleToUser(developerSession.developer.userId))
+      val visibleApis = apmConnector.fetchCombinedApisVisibleToUser(developerSession.developer.userId)
+      updateVisibleApis(developerSession, apmConnector.fetchCombinedApisVisibleToUser(developerSession.developer.userId))
       visibleApis
     }) { x => Future.successful(x.toList) }
   }
 
-  def fetchAllAPICategoryDetails()(implicit hc: HeaderCarrier): Future[List[APICategoryDetails]] = apmConnector.fetchAllAPICategories()
+  def fetchAllAPICategoryDetails()(implicit hc: HeaderCarrier): Future[List[APICategoryDisplayDetails]] = apmConnector.fetchAllAPICategories()
 
-  def apiCategoryDetails(category: String)(implicit hc: HeaderCarrier): Future[Option[APICategoryDetails]] =
+  def apiCategoryDetails(category: String)(implicit hc: HeaderCarrier): Future[Option[APICategoryDisplayDetails]] =
     fetchAllAPICategoryDetails().map(_.find(_.category == category))
 
-  def fetchAPIDetails(apiServiceNames: Set[String])(implicit hc: HeaderCarrier): Future[List[ExtendedApiDefinition]] =
+  def fetchAPIDetails(apiServiceNames: Set[String])(implicit hc: HeaderCarrier): Future[List[CombinedApi]] =
     Future.sequence(
       apiServiceNames
-        .map(apmConnector.fetchAPIDefinition(_))
+        .map(apmConnector.fetchCombinedApi(_))
         .toList)
   
   def fetchEmailPreferencesFlow(developerSession: DeveloperSession) =
-    flowRepository.fetchBySessionIdAndFlowType[EmailPreferencesFlow](developerSession.session.sessionId, FlowType.EMAIL_PREFERENCES) map {
+    flowRepository.fetchBySessionIdAndFlowType[EmailPreferencesFlowV2](developerSession.session.sessionId, FlowType.EMAIL_PREFERENCES) map {
```

Comment:
Shouldn't this be EMAIL_PREFERENCES_V2 to avoid trying to read old values (even if they only last 15 mins).

Reply:
done